iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -309,14 +311,14 @@ template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename T>
 inline constexpr uint64_t storable_function<Capacity, signature<ReturnType, Args...>>::required_storage_size() noexcept
 {
-    return StorageType::template allocation_size<T>();
+    return sizeof(T) + alignof(T) - 1;
```

Comment:
This could be made more exact if the alignment of storage is taken in account (may safe a little memory). But this worst case estimate is correct, just not tight in general. It assumes that we need to maximally realign by `alignof(T) - 1`, but generally it will be less.

It would look like the `align` method we refactored in helplets to get an aligned address from the start of storage and the look whether there is space for `sizeof(T)` from this address onward.

This is why it was previously part of the storage and hence knew about the alignment directly (and another reason for the abstraction). I consider reworking the storage concepts in the future to better reflect this, as it existed before it was functional but not fully developed.

TLDR: can be optimized but does not have to be right now. A change would need to take the starting address of `m_storage` into account.